### PR TITLE
improve dig_node

### PR DESCRIPTION
**Pull request by Desour**

### CODE REVIEW

File: maidroid/api.lua [hunk 1]
```diff
@@ -51,6 +51,15 @@ end
 -- this table's methods must be put there.
 maidroid.maidroid = {}
 
+-- for the case, the luaentity is used as player
+function maidroid.maidroid.is_player(self)
+	return true
```
Comment (HybridDog) [inline maidroid/api.lua]: if maidroid.maidroid.is_player_currently then
maidroid.maidroid.is_player_currently = false
return true
end
return false

File: maidroid_core/cores/ocr.lua [hunk 2]
```diff
@@ -262,12 +262,12 @@ local maidroid_instruction_set = {
 			return true, false, "insufficient tool capabilities"
 		end
 
-		-- dig node
-		def.on_dig(pos, node, obj)
-		--The block not being air is considered "failure".
-		if minetest.get_node(pos).name ~= "air" then
-			return true, false, "no air after digging"
-		end
```
Comment (HybridDog) [inline maidroid_core/cores/ocr.lua:270]: maidroid.maidroid.is_player_currently = true

**Comment by Desour** [comment]
Requested change done.
Problematically the maidroid can't use a tool, I'll change that.

**Comment by Desour** [comment]
Ok, now it's good.

**Pull request merged**